profiles/common/modules/custom/tmgmt_poetry/src/tmgmt_poetry.behat.inc

```diff
@@ -117,6 +117,76 @@ class TmgmtPoetrySubContext extends RawDrupalContext implements DrupalSubContext
   }
 
   /**
+   * Storing job reference from the Page.
+   *
+   * @Then /^I store job Reference of translation request page$/
+   */
+  public function iStoreJobReferenceOfTranslationRequestPage() {
+    $selector = ".messages";
+    $begin_message = "Status message Job has been successfully submitted for translation. Project ID is:";
+    $selector_objects = $this->getSession()->getPage()->findAll("css", $selector);
```

Comment:
Instead scraping the project ID from the user interface (there might be more messages at a certain point, so this might get mixed results) we could maybe refer to the database.

If we set a break at:

```gherkin
...
    And I check the box "settings[languages][it]"
    And I press "Submit to translator"
      And break
```

Then the last row you get in the `tmgmt_message` table is:

```
29	16	NULL	14	Job has been successfully submitted for translation. Project ID is: %project_id	a:1:{s:11:"%project_id";s:37:"MAIN_16_POETRY_ABCD/2016/1234/0/3/TRA";}	1470309192	status
```

Which is still an empirical way of getting the submitted project ID but it's a little more stable that parsing the HTML.